Source/Python/WarpXWrappers.cpp

```diff
@@ -132,6 +132,14 @@ extern "C"
         return PIdx::nattribs;
     }
 
+    int warpx_nCompsSpecies(const char* char_species_name)
+    {
+        auto & mypc = WarpX::GetInstance().GetPartContainer();
+        const std::string species_name(char_species_name);
+        auto & myspc = mypc.GetParticleContainerFromName(species_name);
+        return myspc.NumRealComps();
```

Comment:
See above in the Python part.
In case we only want the runtime added components of type real, that's the lengths of `pc->NumRuntimeRealComps()`.
We can also return both compile-time added and runtime-added components as in #2126